### PR TITLE
feat: added stack function to paddle tensor class

### LINKED ISSUE #23497: stack

### PR DESCRIPTION
**Pull request by arshPratap**
# PR Description
Adds stack function to paddle tensor class

## Related Issue

Close #23497

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by bipinKrishnan** [comment]
ivy-gardener

**Comment by arshPratap** [comment]
Hi @bipinKrishnan ... no `failing_tests` here..could you take a look ?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 1]
```diff
@@ -3330,6 +3330,55 @@ def test_paddle_tensor_squeeze_(
     )
 
 
+# stack
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="stack",
+    dtype_value=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric").filter(
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:3339]: Couldn't we just use the supported dtypes decorator on the function instead of doing the filtering here?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 2]
```diff
@@ -3330,6 +3330,55 @@ def test_paddle_tensor_squeeze_(
...
+    axis,
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    frontend,
+    on_device,
+    backend_fw,
+):
+    input_dtype, data = dtype_value
+    x = Tensor(data[0])
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:3362]: Explicit type casting shouldn't be done here. The value of data should be an array if I'm correct.

**Comment by bipinKrishnan** [comment]
Hey @arshPratap, could you resolve the conflicts here. I found some extra logic being applied at some parts of the test function added, that shouldn't be the case. You could may be refer to other test cases before making the changes. Thanks!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py (hunk 1 above)
Comment (arshPratap) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py]: Resolved @bipinKrishnan ..Thanks for the review

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py (hunk 2 above)
Comment (arshPratap) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py]: @bipinKrishnan  Actually it requrires a list of paddle `Tensor` .. As defined [here](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/stack_en.html#stack) , the right way to provide input to the stack function..I have written a custom function to generate the necessary input for the tests

**Comment by arshPratap** [comment]
> Hey @arshPratap, could you resolve the conflicts here. I found some extra logic being applied at some parts of the testing function you added, that shouldn't be the case. You could may be refer to other functions before making the changes. Thanks!

Hi @bipinKrishnan !
Thanks for the review 
Have rebased to the latest changes in the `main` branch and have worked on the suggestions as well..Could you take a look ?

File: ivy/functional/frontends/paddle/manipulation.py [hunk 3]
```diff
@@ -153,6 +153,9 @@ def squeeze(x, axis=None, name=None):
     return ivy.squeeze(x, axis=axis)
 
 
+@with_supported_dtypes(
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/manipulation.py:156]: Since these files are not related to the PR, you could add this decorator in the tensor instance of `stack` function.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 4]
```diff
@@ -152,6 +152,27 @@ def _get_dtype_and_values_for_lerp(draw):
         return input_dtype, x[0], x[1], weight
 
 
+# stack
+@st.composite
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:156]: helper functions should be in their respective folder

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
